client/post-editor/media-modal/detail/detail-item.jsx

```diff
@@ -27,18 +28,47 @@ module.exports = React.createClass( {
 		hasPreviousItem: React.PropTypes.bool,
 		hasNextItem: React.PropTypes.bool,
 		onShowPreviousItem: React.PropTypes.func,
-		onShowNextItem: React.PropTypes.func
+		onShowNextItem: React.PropTypes.func,
+		onEdit: React.PropTypes.func
 	},
 
 	getDefaultProps: function() {
 		return {
 			hasPreviousItem: false,
 			hasNextItem: false,
 			onShowPreviousItem: noop,
-			onShowNextItem: noop
+			onShowNextItem: noop,
+			onEdit: noop
 		};
 	},
 
+	renderEditButton: function() {
+		var mimePrefix, Component;
```

Comment:
I'd personally be fine with using newer ES6 variable declarations in this function, even if it's inconsistent with other functions in the file.